### ISSUE TITLE
[J3.9.20] UX - Help users not to produce 404 Errors

### ISSUE THREAD

**Issue by coolcat-creations** [joomla/joomla-cms#30273]
### Is your feature request related to a problem? Please describe.
I am currently teaching a class of kids in working with Joomla. So here and there always the same mistakes happen where we can learn from. If a new user comes into such situations the frustration can be high. But we can help them not to make such mistakes.

So in current class and also in the kids4opensource initiative in Joomla it happened, that the sites either threw an error 404 or a "too many redirects".

The reasons where (one of)
- Article in the trash
- Category in the trash
- Language disabled

### Describe the solution you'd like

When a user puts something (article or category or category of an article) in a trash which is still connected to a menu item a modal should appear and offer:
- connect to another article
- put the menuitem also in the trash (if it's not the default one)

When a user disables the main language a modal should appear to change the default language

### Additional context
Those kids are the future of Joomla and we should make the work with Joomla as smooth as possible. 
Also it should be a "Don't make me think".

**Comment by softforge** [comment]
It's too late for 3.9 series as there are no new features in3.9.21 but this is really needed for the 4 series and would be cool new features for 4.1

**Comment by coolcat-creations** [comment]
> Its probably too late for 3.9.20 as there is no 3.9.21 but this is really needed for the 4 series and would be cool new features for 4.1

yes I know, but I hope I can add here the issues that happen in beginner classes and that they find attention somewhere. If I'd know how to develop it I would certainly do it :-)

**Comment by brianteeman** [comment]
To achieve this I would look at how a menu item is disabled when the component it is using is uninstalled

**Comment by brianteeman** [comment]
> When a user puts something (article or category or category of an article) in a trash which is still connected to a menu item a modal should appear and offer:

An article doesn't know if it is connected to a menu item so the only way this could be done is on every article save to scan all the menu items and to see if it is a menu item. This obviously would not be an optimal experience especially on a larger site.

> When a user disables the main language a modal should appear to change the default language

I assume that you mean on the extensions:manage page - I cant see anywhere else you could disable a language. The same problem as above. You cant tell on that page if a language is used or is set as default.
